### ISSUE TITLE
Automate reformatting and PEP8 compliance

### ISSUE THREAD

**Issue by andrelmfarias** [cdqa-suite/cdQA#231]
**Is your feature request related to a problem? Please describe.**
Keeping code PEP8 compliant and automatic reformating is something we should assure when we commit / merge to the main branch.

**Describe the solution you'd like**
The solution would be to enable / install black and Flack8 as described here: https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/

I had this idea taken from the github repository of PySyft. 

**Describe alternatives you've considered**
The alternative is to use `flake8-black` (https://pypi.org/project/flake8-black/).

**Comment by fmikaelian** [comment]
I often use a pytest plugin that performs flake8 tests in the CI: https://github.com/tholo/pytest-flake8

We just need to update the following line: https://github.com/cdqa-suite/cdQA/blob/7b80682eddcf6e9b4f92371123e2ff79c6304e30/.travis.yml#L8

**Comment by andrelmfarias** [comment]
yes, but it would be useful to require contributors to use black -> it reformats automatically the code by running `black . `. It is very useful when we write huge codes...

**Comment by andrelmfarias** [comment]
We can add a sort of requirements for devs, which include pre-commit hooks. Such as OpenMined does with pysyft: https://github.com/OpenMined/PySyft/blob/dev/requirements_dev.txt
